src/App.jsx

```diff
@@ -1,14 +1,86 @@
+import { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
+import messageData from './data/messages.json';
 
 const App = () => {
+  const [messages, setMessages] = useState(messageData);
+  const [likedCount, setLikedCount] = useState(0);
+  const [remoteColor, setRemoteColor] = useState();
+  const [localColor, setLocalColor] = useState();
+
+  const toggleLikedMessage = (messageId) => {
+    const updatedMessages = messages.map((message) => {
+      return message.id === messageId
+        ? { ...message, liked: !message.liked }
+        : message;
+    });
```

Comment:
Nice work on this function! By passing this down to your individual chat messages you are now able to have a single source of truth!